pkg/sql/doctor/doctor_test.go

```diff
@@ -60,8 +60,28 @@ var validTableDesc = &descpb.Descriptor{
 	},
 }
 
-func toBytes(t *testing.T, pb protoutil.Message) []byte {
-	res, err := protoutil.Marshal(pb)
+func toBytes(t *testing.T, desc *descpb.Descriptor) []byte {
+	if desc.GetDatabase() != nil {
+		if desc.GetDatabase().Privileges == nil {
+			descpb.MaybeFixPrivileges(desc.GetDatabase().GetID(), &desc.GetDatabase().Privileges)
+		}
+	} else if desc.GetSchema() != nil {
+		if desc.GetSchema().Privileges == nil {
+			descpb.MaybeFixPrivileges(desc.GetSchema().GetID(), &desc.GetSchema().Privileges)
+		}
+	} else if tbl := descpb.TableFromDescriptor(desc, hlc.Timestamp{}); tbl != nil {
+		if tbl.Privileges == nil {
+			descpb.MaybeFixPrivileges(tbl.GetID(), &tbl.Privileges)
+		}
+		if tbl.FormatVersion == 0 {
+			tbl.FormatVersion = descpb.InterleavedFormatVersion
+		}
+	} else if typ := descpb.TypeFromDescriptor(desc, hlc.Timestamp{}); typ != nil {
+		if typ.Privileges == nil {
+			descpb.MaybeFixPrivileges(typ.GetID(), &typ.Privileges)
+		}
+	}
+	res, err := protoutil.Marshal(desc)
```

Comment:
Note: I really couldn't be bothered to update all of the descriptor declarations so I did this instead.